src/utils.rs

```diff
@@ -125,9 +160,61 @@ pub(crate) fn handle_vault_path_argument(path: Option<String>) -> Result<PathBuf
     Ok(vault_path)
 }
 
+/// Encrypts and saves the mnemonic phrase to disk
+pub(crate) fn save_phrase_to_disk<P>(vault_path: &P, mnemonic: &str, password: &str)
+where
+    P: AsRef<Path> + Debug,
```

Comment:
While the use of `P: AsRef<Path>` is pretty common in the `std` library, it's widely considered a bit of a mistake. This is because the generic path argument means that the function requires monomorphisation, which tends to result in a lot more code generation which can slow down build times and bloat binaries for very little usability gain.

In other words, if the function is called with both a `&str` and `&Path` the entire function will be generated twice, even though the `str` argument could have just been passed with `.as_ref()` on the caller side!

In general, it's better to just use `&Path` and resolve the generic on the caller side, e.g. `save_phrase_to_disk(&path, mnemonic, pw)` or in the case that `path` is a `str`, `save_phrase_to_disk(path.as_ref(), mnemonic, pw)`.